### PR TITLE
Moved api pkg from eventing-controller

### PR DESCRIPTION
**Pull request by mfaizanse**
**Description**

Changes proposed in this pull request are mentioned [here](https://github.com/kyma-project/eventing-manager/issues/106#issuecomment-1819383999).

**Related issue(s)**

- https://github.com/kyma-project/eventing-manager/issues/106

### CODE REVIEW

File: pkg/env/nats_config_test.go
Comment (mfaizanse) [inline pkg/env/nats_config_test.go:35]: This is done to fix import cycle!!!

File: docs/user/02-configuration.md
Comment (mfaizanse) [inline docs/user/02-configuration.md:31]: Revert this file to version from main branch before getting approval.

**Review by muralov — DISMISSED** [review]

File: docs/user/02-configuration.md
Comment (IwonaLanger) [reply docs/user/02-configuration.md:31]: Agree :)

**Review by IwonaLanger — APPROVED** [review]

**Review by muralov — APPROVED** [review]

**Review by NHingerl — APPROVED** [review]

**Pull request merged**
